mne/beamformer/_compute_beamformer.py

```diff
@@ -116,7 +116,25 @@ def _sym_inv(x, reduce_rank):
     return np.matmul(u * s[:, np.newaxis], u.transpose(0, 2, 1))
 
 
-def _normalized_weights(Wk, Gk, Cm_inv_sq, reduce_rank, nn, sk):
+def _reduce_leadfield_rank(G):
+    """Reduce the rank of the leadfield."""
+    # decompose lead field
+    u, s, v = np.linalg.svd(G)
+    s[:, np.argmin(s, axis=1)] = 0.  # set the smallest singular value to 0.
+
+    # expand s to match dimension of u
+    s_full = np.zeros(G.shape)
+    # TODO: vectorize?
```

Comment:
Don't actually construct these diagonal matrices. Instead, promote `s`'s dimensionality and operate elementwise using broadcasting (this is equiv to matmul with a diag mat) like:
```
s = s[..., np.newaxis, :]
G = np.matmul(u * s, v)
```